test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +170,92 @@ describe('View existing record,', function() {
         });
     });
 
+    describe("For side panel table of contents in Record App", function() {
+
+        beforeAll(function() {
+            var url = browser.params.url + "/record/#" + browser.params.catalogId + "/" + testParams.sidePanelTest.schemaName + ":" + testParams.sidePanelTest.tableName +  "/id=" + testParams.sidePanelTest.id;
+            browser.get(url);
+            recSidePan = chaisePage.recordPage.getElementById('recordSidePan-5');
+            fiddlerBtn = element(by.className('sidePanFiddler')).element(by.tagName('i')),
+            chaisePage.waitForElement(fiddlerBtn);
+        });
+        it('Table of contents should be collapsed by default', function(){
+            var recordSidePan = chaisePage.recordPage.getElementById('record-side-pan');
+            recordSidePan.isDisplayed().then(function (bool) {
+                expect(bool).toBeFalsy();
+            });
+        });
+
+        it('Table of contents should match all related table', function(){
+            var  allPan = element.all(by.xpath("//div[contains(@id,'recordSidePan')]"));
+            chaisePage.waitForElement(fiddlerBtn).then(function(){
+              return fiddlerBtn.click();
+            }).then(function(){
+              expect(allPan.count()).toBe(testParams.sidePanelTest.tocCount, "Table of contents count did not match!");
+            }).catch( function(err) {
+               console.log(err);
+           });
+        });
+
+        it('On click of Related table name in TOC, page should move to the contents and open the table details', function(){
+            var rtTableHeading = chaisePage.recordPage.getRelatedTableHeading(testParams.sidePanelTest.tableToShow),
+                lastRow = element(by.id('rt-heading-Categories_5')).all(by.tagName('tr')).get(5);
+
+            recSidePan.getAttribute('class').then(function(className) {
+                return recSidePan.click();
+            }).then (function (){
+              // related table should be visible
+                rtTableHeading.isDisplayed().then(function (bool) {
+                  expect(bool).toBeTruthy();
+                });
+                chaisePage.waitForElement(lastRow);
+                return rtTableHeading.getAttribute("class");
+            }).then (function(className) {
+                expect(className).toContain("panel-open", "Related table panel is not open when clicked through TOC.");
+            }).catch( function(err) {
+                console.log(err);
+            });
+        });
+
+        it('Record count should match for pan and related table content', function(){
```

Comment:
All you need to test in this test case is if the side panel is displaying the proper count. We already know what the count should be (it should be defined as part of our expected data at the top of the spec). You do not need to count the rows and verify it's represented by the same value in the text of the side panel.

Simply grab the heading you are verifying and make sure the proper text is displayed.

Reply:
Now that I think about it more, you should verify the full list of headings, you should be making sure each heading shows up properly and in the right order. [See here](https://github.com/informatics-isi-edu/chaise/blob/master/test/e2e/utils/record-helpers.js#L180-L185).